apps/system/test/unit/bluetooth_transfer_test.js

```diff
@@ -813,5 +827,60 @@ suite('system/bluetooth_transfer', function() {
         });
       });
     });
+
+    suite('handleSystemMessageNotification behavior', function() {
+      test('calls openReceivedFile', function() {
+        var message = {
+          body: 'someFile.txt',
+          data: {
+            systemMessageTarget: 'BluetoothTransfer',
+            contentType: '',
+          }
+        };
+        this.sinon.spy(BluetoothTransfer, 'openReceivedFile');
+        this.sinon.spy(BluetoothTransfer, 'closeSystemMessageNotification');
+        BluetoothTransfer.handleSystemMessageNotification(message);
+
+        assert.isTrue(BluetoothTransfer.openReceivedFile.calledOnce);
+        assert.isTrue(
+          BluetoothTransfer.closeSystemMessageNotification.calledOnce);
+      });
+    });
+
+    suite('closeSystemMessageNotification behavior', function() {
+      var notifCloseSpy, notificationGetStub;
+      var notification = {
+        body: 'fake',
+        data: {
+          systemMessageTarget: 'BluetoothTransfer'
+        },
+        close: function() {}
+      };
+
+      setup(function() {
+        notifCloseSpy = this.sinon.spy(notification, 'close');
+        notificationGetStub = function notificationGet() {
+          return {
+            then: function(cb) {
+              cb && cb([ notification ]);
+            }
+          };
+        };
+        this.sinon.stub(window.Notification, 'get', notificationGetStub);
+      });
+
+      test('closes notification by body', function() {
```

Comment:
refer to https://github.com/mozilla-b2g/gaia/blob/master/apps/system/test/unit/findmydevice_launcher_test.js#L280